### ISSUE TITLE
Pyright doesn't allow overriding methods with polymorphic functions

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#9417]
```python
class A:
    def f(self, v: str) -> str: ...

class B(A):
    def f[T](self, v: T) -> T: ...  # "object*" is not assignable to "str"
```

The above is expected to pass type checking. Pyright allows assigning `(T@f) -> T@f` to `(str) -> str`, so there should be no problems with these functions being methods.

Related to https://github.com/microsoft/pyright/issues/9347.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.389
